enclave/core/sgx/globals.c

```diff
@@ -189,9 +190,29 @@ size_t __oe_get_reloc_size()
 **==============================================================================
 */
 
-const void* __oe_get_eeid()
+const oe_eeid_page_t* __oe_get_eeid_page()
 {
-    return oe_eeid;
+    if (!_eeid_rva)
+        return NULL;
+
+    const unsigned char* base = __oe_get_enclave_base();
+    return (oe_eeid_page_t*)(base + _eeid_rva);
+}
+
+const oe_eeid_t* __oe_get_eeid()
+{
+    if (!_eeid_rva)
+        return NULL;
+
+    const oe_eeid_page_t* eeid_page = __oe_get_eeid_page();
+    return eeid_page && eeid_page->eeid.version == OE_EEID_VERSION
```

Comment:
In the updated EEID design, the enclave code shoud check `entry_rva` in the eeid data matches the value in oeprops.